client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -23,46 +23,36 @@ import {
  */
 const log = debug( 'calypso:middleware-media' );
 
-/**
- * Issues an API request to fetch media for a site and query.
- *
- * @param  {Object}  store  Redux store
- * @param  {Object}  action Action object
- */
-export function requestMedia( { dispatch }, action ) {
-	dispatch( requestingMedia( action.siteId, action.query ) );
-
+export function requestMedia( action ) {
 	log( 'Request media for site %d using query %o', action.siteId, action.query );
 
-	dispatch(
+	return [
+		requestingMedia( action.siteId, action.query ),
 		http(
 			{
 				method: 'GET',
 				path: `/sites/${ action.siteId }/media`,
 				apiVersion: '1.1',
 			},
 			action
-		)
-	);
+		),
+	];
 }
 
-export function requestMediaSuccess( { dispatch }, { siteId, query }, { media, found } ) {
-	dispatch( receiveMedia( siteId, media, found, query ) );
-	dispatch( successMediaRequest( siteId, query ) );
-}
+export const requestMediaSuccess = ( { siteId, query }, { media, found } ) => [
+	receiveMedia( siteId, media, found, query ),
+	successMediaRequest( siteId, query ),
+];
 
-export function requestMediaError( { dispatch }, { siteId, query } ) {
-	dispatch( failMediaRequest( siteId, query ) );
-}
+export const requestMediaError = ( { siteId, query } ) => failMediaRequest( siteId, query );
 
-export function handleMediaItemRequest( { dispatch }, action ) {
+export function handleMediaItemRequest( action ) {
 	const { mediaId, query, siteId } = action;
 
-	dispatch( requestingMediaItem( siteId, query ) );
-
 	log( 'Request media item %d for site %d', mediaId, siteId );
 
-	dispatch(
+	return [
+		requestingMediaItem( siteId, query ),
```

Comment:
This should be `requestingMediaItem( siteId, mediaId )`, but it was broken before this PR:

https://github.com/Automattic/wp-calypso/blob/17276793273dd19f31a6e52de675fb8222918ff4/client/state/media/actions.js#L129-L139

Reply:
🤦‍♂️ I believe that is an error when this was implemented alongside #18022. The functions have nearly identical signatures.

Ideally we could remove all of this is requesting logic as noted here https://github.com/Automattic/wp-calypso/pull/18017/files/e548beb47d31ecb3945ddb45d9f3a89065b85e80#diff-8a834454d3ca55002e933663ede8d986, but none of these are blockers for this PR.

Reply:
fixed in 912aeae

Reply:
I take it back - that led to test failures so for this PR I'm not going to change it